ir_datasets/datasets/hc4.py

```diff
@@ -0,0 +1,168 @@
+from typing import Dict, NamedTuple
+import json
+import ir_datasets
+from ir_datasets.formats.base import BaseDocs, BaseQueries
+from ir_datasets.indices.lz4_pickle import PickleLz4FullStore
+from ir_datasets.util import DownloadConfig
+from ir_datasets.formats import TrecQrels
+from ir_datasets.datasets.base import Dataset, YamlDocumentation
+
+NAME = 'hc4'
+
+LANG_CODE_CONVERT = {
+    'zh': 'zho',
+    'fa': 'fas',
+    'ru': 'rus'
+}
+
+DOC_COUNTS = {
+    'zh': 646305,
+    'fa': 486486,
+    'ru': 4721064
+}
+
+class HC4Doc(NamedTuple):
+    doc_id: str
+    title: str
+    text: str
+    url: str
+    time: str
+    cc_file: str
+
+
+class HC4Docs(BaseDocs):
+    
+    def __init__(self, docs_dlc, subset_lang):
+        self._docs_dlc = docs_dlc
+        self._subset_lang = subset_lang
+        self._count = DOC_COUNTS[subset_lang]
+
+    def docs_path(self, force=True):
+        return self._docs_dlc.path(force)
+
+    @ir_datasets.util.use_docstore
+    def docs_iter(self):
+        with self._docs_dlc.stream() as f:
+            for line in f:
+                line = json.loads(line)
+                line['doc_id'] = line['id']
+                del line['id']
+                yield HC4Doc(**line)
+
+    def docs_store(self):
+        return PickleLz4FullStore(
+            path=f'{self.docs_path(force=False)}.pklz4',
+            init_iter_fn=self.docs_iter,
+            data_cls=self.docs_cls(),
+            lookup_field='doc_id',
+            index_fields=['doc_id'],
+            count_hint=self._count,
+        )
+
+    def docs_cls(self):
+        return HC4Doc
+
+    def docs_namespace(self):
+        return NAME
+
+    def docs_count(self):
+        return self._count
+
+    def docs_lang(self):
+        return self._subset_lang 
+
+
+class HC4Query(NamedTuple):
+    query_id: str
+    title: str
+    description: str
+    ht_title: str
+    ht_description: str
+    mt_title: str
+    mt_description: str
+    narrative: Dict[str, str]
```

Comment:
should be `str`?

Reply:
We actually provided one example per relevance level in the narrative field. 
I don't think there will be a lot of people using it as a query so decided to use a structure that's less common here.

Reply:
Ah, gotcha! That's really nice, actually. Maybe it's worth renaming then so it's clear what the keys/values mean. E.g., `narrative_by_relevane`?

Reply:
Ah that makes sense!

Reply:
and done :)